server/src/main/java/org/apache/iotdb/db/sync/sender/manager/LocalSyncManager.java

```diff
@@ -46,8 +53,30 @@ public LocalSyncManager(DataRegion dataRegion, Pipe pipe) {
 
   /** tsfile */
   @Override
-  public void syncRealTimeDeletion(Deletion deletion) {
-    syncPipe.collectRealTimeDeletion(deletion, dataRegion.getStorageGroupName(), dataRegionId);
+  public void syncRealTimeDeletion(
+      Deletion deletion,
+      List<TsFileResource> unsealedResources,
+      List<TsFileResource> sealedResources) {
+    Set<String> devices = new HashSet<>();
+    unsealedResources.forEach(o -> devices.addAll(o.getDevices()));
+    sealedResources.forEach(o -> devices.addAll(o.getDevices()));
+
+    for (String device : devices) {
+      try {
+        PartialPath path = new PartialPath(device);
+        if (deletion.getPath().overlapWith(path)) {
+          Deletion splitDeletion =
+              new Deletion(path, 0, deletion.getStartTime(), deletion.getEndTime());
```

Comment:
When deletion.getPath() is a full path without wildcards, the splitting logic may be error.

e.g. "root.sg.d1.s1".overlapWith("root.sg.d1") is false.

Reply:
done.